tools/packaging/common/istio-iptables.sh

```diff
@@ -59,31 +60,61 @@ function dump {
     ip6tables-save
 }
 
-#
-#  Function check if argument is a valid ipv4 or ipv6 address
-#
 function isValidIP() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   if isIPv4 "${1}"; then
       true
-   elif [ "$1" != "${1#*:[0-9a-fA-F]}" ]; then
+   elif isIPv6 "${1}"; then
       true
    else
       false
    fi
 }
-
 #
 # Function return true if agrument is a valid ipv4 address
 #
 function isIPv4() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   local ipv4regexp="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$"
+   if [[ ${1} =~ ${ipv4regexp} ]]; then
       true
    else
       false
   fi
 }
-
-trap dump EXIT
+#
+# Function return true if agrument is a valid ipv6 address
+#
+function isIPv6() {
+  local ipv6section="^[0-9a-fA-F]{1,4}$"
+  addr="$1"
+  number_of_parts=0
+  number_of_skip=0
+  IFS=':' read -r -a addr <<< "$1"
+  if [ ${#addr[@]} -eq 0 ]; then
```

Comment:
Can we simplify this ? It's really just v6 or v4 - and the presence of a : is sufficient. 
We can assume the OS returns a valid IP - again, we didn't see any problems before having a validation
for v4.

Reply:
If user specifies incorrect include/exclude ranges then it will cause failure. You did not see problems before because it is not yet widely used and there is probably one or two tests specifying include/exclude ranges in pod's yaml. Having good sanity checks will protect from unexplainable crash loops.

Reply:
If we decide to officially support this ( right now it's a pretty experimental annotation - not part of the official API ) - Galley can validate as much as it needs. When we switch to CNI as sidecar and possibly reimplement this in go - we can do validation ( assuming this annotation can be supported - since we may switch off iptables and use veth pair ). 

If you feel it's important enough - you can probably add a validation on the Pod object in Galley and reject it. 

Don't forget that by the time iptable.sh is run - the user has no way to fix the problem even if validation fails. So one way or another - things will not work - either crash loop or ignored config is bad.

Reply:
If you could point me to galley code, I would be happy to add validation and then retire bash validation.
Also I can refactor istio-iptables.sh in go if you are ok with it.